databuilder/query_model.py

```diff
@@ -91,10 +95,6 @@ def __repr__(self):
 class Node:
     "Abstract base class for all objects in the Query Model"
 
-    def __init_subclass__(cls, **kwargs):
-        # All nodes in the query model are frozen dataclasses
-        dataclasses.dataclass(cls, frozen=True)
-
```

Comment:
As per my commit message, I'm not entirely happy about this change. I'm open to reverting it as long as we've discussed the tradeoffs.

Reply:
Yes, I do see the particular value in being able to typecheck dataset definitions so it makes sense to be willing to pay a higher price there — but ideally we wouldn't have to pay it everywhere.